### PR TITLE
hosted_registry: clean up tmp mount point and fstab

### LINKED ISSUE #6004: gluster: temporary mountpoint gets left behind for registry
#### Description

After doing an install where I have a registry backed by glusterfs I end up with a failing systemd mount unit on every reboot of my master node: 

```
● sysroot-tmp-openshift\x2dglusterfs\x2dregistry\x2d5ufMjT.mount - /sysroot/tmp/openshift-glusterfs-registry-5ufMjT
   Loaded: loaded (/etc/fstab; generated; vendor preset: disabled)
   Active: failed (Result: exit-code) since Fri 2017-11-03 13:13:51 UTC; 2h 34min ago
    Where: /sysroot/tmp/openshift-glusterfs-registry-5ufMjT
     What: 45.55.58.69:/glusterfs-registry-volume
     Docs: man:fstab(5)
           man:systemd-fstab-generator(8)
  Process: 1120 ExecMount=/usr/bin/mount 45.55.58.69:/glusterfs-registry-volume /sysroot/tmp/openshift-glusterfs-registry-5ufMjT -t glusterfs (code=exited, status=1/FAILURE)

Nov 03 13:13:51 origin-master-1.localdomain systemd[1]: Mounting /sysroot/tmp/openshift-glusterfs-registry-5ufMjT...
Nov 03 13:13:51 origin-master-1.localdomain systemd[1]: sysroot-tmp-openshift\x2dglusterfs\x2dregistry\x2d5ufMjT.mount: Mount process exited, code=exited status=1
Nov 03 13:13:51 origin-master-1.localdomain systemd[1]: Failed to mount /sysroot/tmp/openshift-glusterfs-registry-5ufMjT.
Nov 03 13:13:51 origin-master-1.localdomain systemd[1]: sysroot-tmp-openshift\x2dglusterfs\x2dregistry\x2d5ufMjT.mount: Unit entered failed state.
```

Looks like it is coming from https://github.com/openshift/openshift-ansible/blob/c99bfde17bca322e8a89ba3e9dc4e4fdc2776448/roles/openshift_hosted/tasks/storage/glusterfs.yml#L28-L39 

Is this only a temporary mount that is needed during install or is this something that needs to be persistent. Based on the answer to that we should either make sure the systemd mount unit gets cleaned up or set up the mountpoint to be somewhere more persistent than `/tmp/`. 

##### Version

Openshift-ansible @ a7a32bb targeting cluster of 3 fedora 27 atomic hosts. 

Similar inventory setup as what was provided in https://github.com/openshift/openshift-ansible/issues/5935#issue-269704409

### PR DESCRIPTION
**Pull request by dustymabe**
Fixes #6004

### CODE REVIEW

File: roles/openshift_hosted/tasks/storage/glusterfs.yml [hunk 1]
```diff
@@ -79,14 +79,7 @@
       - REGISTRY_STORAGE_MAINTENANCE_READONLY_ENABLED: 'true'
   when: openshift.hosted.registry.storage.glusterfs.swap
 
-- name: Unmount registry volume
+- name: Unmount registry volume and clean up mount point/fstab
   mount:
-    state: unmounted
-    name: "{{ mktemp.stdout }}"
-
-- name: Delete temp mount directory
```
Comment (michaelgugino) [inline roles/openshift_hosted/tasks/storage/glusterfs.yml]: If we no longer need this temp directory, let's clean it up.

Comment (dustymabe) [reply roles/openshift_hosted/tasks/storage/glusterfs.yml]: `absent` in the mount module does this for us

**Comment by dustymabe** [comment]
any tips on the test failures?

File: roles/openshift_hosted/tasks/storage/glusterfs.yml (hunk 1 above)
Comment (michaelgugino) [reply roles/openshift_hosted/tasks/storage/glusterfs.yml]: Looks like you're right.  But it also looks like using 'state: absent' will not actually unmount the mount according to the docs:

'absent and present only deal with fstab but will not affect current mounting.'

Looks like we need two statements here.

**Comment by michaelgugino** [comment]
> any tips on the test failures?

We're having some CI issues at the moment.  These failures are not related to the PR.  I'll kick the tests again once this clears.

File: roles/openshift_hosted/tasks/storage/glusterfs.yml (hunk 1 above)
Comment (dustymabe) [reply roles/openshift_hosted/tasks/storage/glusterfs.yml]: yeah that's what I thought too. and then we found a trail of "wrong". It starts [here](https://github.com/openshift/openshift-ansible/issues/6004#issuecomment-342542201).

**Comment by dustymabe** [comment]
@michaelgugino - mind doing the honors? would like to get this in

**Comment by michaelgugino** [comment]
@dustymabe if install passes, I'll merge it.

**Comment by jlebon** [comment]
bot, retest this please

**Comment by dustymabe** [comment]
the test failures from https://s3.amazonaws.com/aos-ci/ghprb/openshift/openshift-ansible/faf425ca8a33a24168da808fc22667d3158bff35.0.1510337116615656111/index.html actually look to be: 
https://github.com/openshift/openshift-ansible/issues/6086

**Comment by dustymabe** [comment]
```
    <string>: container_linux.go:247: starting container process caused "process_linux.go:258: applying cgroup configuration for process caused \"open /sys/fs/cgroup/pids/kubepods.slice/kubepods-besteffort.slice/kubepods-besteffort-pod97fcd60d_ca1d_11e7_bd1e_0efc5783583a.slice/docker-4dd6c549447dc96a815b69f8db1bc16cf122c262d180092077a9551db1559118.scope/cgroup.procs: no such file or directory\""
```

I don't know how to interpret that result?

**Comment by michaelgugino** [comment]
@dustymabe I was referring to the ci/openshift-jenkins/* jobs.  Specifically, the 'install' job.  We are in the process of cleaning up/fxing some CI tests that broke after tagging the newest release.

**Comment by dustymabe** [comment]
looks like 'install' passed

**Pull request merged**